pkgs/sketch_pad/web/frame.js

```diff
@@ -22,16 +22,23 @@ function messageHandler(e) {
   var obj = e.data;
 
   if (obj.command === 'execute') {
-    // TODO: Switch to using engineInitializer.initializeEngine(config). See
-    // https://docs.flutter.dev/development/platform-integration/web/initialization.
-    window.flutterConfiguration = {
-      canvasKitBaseUrl: obj.canvasKitBaseUrl
-    };
-
-    replaceJavaScript(obj.js);
+    runFlutterApp(obj.js);
   }
 };
 
+function runFlutterApp(compiledScript) {
+  var blob = new Blob([compiledScript], {type: 'text/javascript'});
+  var url = URL.createObjectURL(blob);
+  _flutter.loader.loadEntrypoint({
+    entrypointUrl: url,
+    onEntrypointLoaded: function(engineInitializer) {
+      engineInitializer.initializeEngine().then(function(appRunner) {
+        appRunner.runApp();
+      });
+    }
```

Comment:
```suggestion
    onEntrypointLoaded: async function(engineInitializer) {
      let appRunner = await engineInitializer.initializeEngine();
      appRunner.runApp();
    }
```

You can write this with `async/await` in JS and it turns out a little bit prettier IMO.